### DISCUSSION TITLE
npm run build suddenly results in "Can't resolve ffmpeg-static"

### DISCUSSION THREAD

**Discussion by Glutch** [vercel/next.js#23441] [Help]
npm run build results in

"ModuleNotFoundError: Module not found: Error: Can't resolve 'ffmpeg-static' in '/Users/glutch/Documents/dev/javascript/xxx/node_modules/prism-media/src/core'"

Recently i just added markdown-it, then it started crashing. But removing markdown-it, node_modules and re-installing makes no difference. I have no idea what ffmpeg-static or "prism-media" is.

How can i solve this?

These are my dependencies

```
"dependencies": {
    "@sendgrid/mail": "^7.4.2",
    "discord.js": "^12.5.1",
    "markdown-it": "^12.0.4",
    "moment": "^2.29.1",
    "mongoose": "^5.11.20",
    "next": "10.0.8",
    "next-auth": "^3.11.2",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "styled-components": "^5.2.1"
  },
  "peerOptionalDependencies": {
    "mongodb": "^3.6.0"
  },
  "devDependencies": {
    "babel-plugin-styled-components": "^1.12.0"
  }
```

**Comment by JacobLey** [comment]
It looks like `discord.js` is your culprit, or at least the place to start looking

https://www.npmjs.com/package/prism-media Prism Media is a video transformer, that discord uses internally https://github.com/discordjs/discord.js/blob/master/package.json#L52

https://www.npmjs.com/package/ffmpeg-static is a low-level video library that prism media uses internally: https://github.com/amishshah/prism-media/blob/main/package.json#L41

You can see the line in question in the core library: https://github.com/amishshah/prism-media/blob/75951648c5813ae10972c5d38a3cbb4f355d16bb/src/core/FFmpeg.js#L126

If discord.js is being bundled client-side, webpack will try to resolve that file and will end up failing. The solution there is to alias/ignore/resolve to an empty object with webpack.

If the file is being included server-side only, I suspect you are somehow instantiating that FFmpeg process right away and triggering the error
